### ISSUE TITLE
BUG: The behavior for `DataFrame.corrwith` is changed in pandas 1.5.0 ??

### ISSUE THREAD

**Issue by itholic** [pandas-dev/pandas#48826]
### Pandas version checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pandas.pydata.org/docs/whatsnew/index.html) of pandas.

- [X] I have confirmed this bug exists on the main branch of pandas.

### Reproducible Example

```python
import pandas as pd
import numpy as np

df1 = pd.DataFrame(
    {"A": [1, np.nan, 7, 8], "B": [False, True, True, False], "C": [10, 4, 9, 3]}
)
df2 = df1[["B", "C"]]
(df1 + 1).corrwith(df2.B, method="spearman")
```

### Issue Description

Seems like the behavior of `pandas.DataFrame.corrwith` is changed from pandas 1.5.0:

The result from the reproducible example in pandas 1.4.0:
```python
>>> (df1 + 1).corrwith(df2.B, method="spearman")
A    0.0
B    1.0
C    0.0
dtype: float64
```
However, the result from the reproducible example in pandas 1.5.0:
```python
>>> (df1 + 1).corrwith(df2.B, method="spearman")
A    0.5
B    1.0
C   -0.2
dtype: float64
```
I see the https://github.com/pandas-dev/pandas/pull/46174 that improve the performance, but maybe the behavior also changed?

### Expected Behavior

The expected behavior in the previous pandas was:
```python
>>> (df1 + 1).corrwith(df2.B, method="spearman")
A    0.0
B    1.0
C    0.0
dtype: float64
```

### Installed Versions

<details>

INSTALLED VERSIONS
------------------
commit           : 87cfe4e38bafe7300a6003a1d18bd80f3f77c763
python           : 3.9.11.final.0
python-bits      : 64
OS               : Darwin
OS-release       : 21.6.0
Version          : Darwin Kernel Version 21.6.0: Wed Aug 10 14:25:27 PDT 2022; root:xnu-8020.141.5~2/RELEASE_X86_64
machine          : x86_64
processor        : i386
byteorder        : little
LC_ALL           : None
LANG             : None
LOCALE           : None.UTF-8

pandas           : 1.5.0
numpy            : 1.22.3
pytz             : 2021.3
dateutil         : 2.8.2
setuptools       : 61.2.0
pip              : 22.2.2
Cython           : 0.29.24
pytest           : 6.2.5
hypothesis       : None
sphinx           : 3.0.4
blosc            : None
feather          : None
xlsxwriter       : None
lxml.etree       : None
html5lib         : None
pymysql          : None
psycopg2         : None
jinja2           : 3.0.3
IPython          : 7.29.0
pandas_datareader: None
bs4              : 4.10.0
bottleneck       : 1.3.4
brotli           : None
fastparquet      : None
fsspec           : None
gcsfs            : None
matplotlib       : 3.5.1
numba            : None
numexpr          : 2.8.1
odfpy            : None
openpyxl         : 3.0.10
pandas_gbq       : None
pyarrow          : 9.0.0
pyreadstat       : None
pyxlsb           : None
s3fs             : None
scipy            : 1.7.2
snappy           : None
sqlalchemy       : 1.4.26
tables           : None
tabulate         : 0.8.9
xarray           : None
xlrd             : None
xlwt             : None
zstandard        : None
tzdata           : None

</details>

**Comment by MarcoGorelli** [comment]
Hey, thanks for the report

> I see the https://github.com/pandas-dev/pandas/pull/46174 that improve the performance, but maybe the behavior also changed?

Yup, [git bisect](https://www.kaggle.com/code/marcogorelli/pandas-regression-example?scriptVersionId=106717506) confirms

**Comment by MarcoGorelli** [comment]
cc @fractionalhare (no blame, just in case you know how to fix it)

**Comment by phofl** [comment]
Reverting would be an option, if we can not figure out what caused this

**Comment by GYHHAHA** [comment]
This is caused by scipy.spearman use [average](https://github.com/scipy/scipy/blob/656076ca6b490f587e9bd9c4cd10cb259a687c5b/scipy/stats/_stats_py.py#L9047) strategy to rank instead of arr.argsort().argsort(). So if the df1's cols have different nan positions, then the k[nonnull_mask].argsort().argsort() in `corrwith` will differ. Actually we still compute the correlation as many times as we use `apply` to do the computing before, thus there is no PERF as far as I can see. +1 on revert this. @MarcoGorelli @phofl

**Comment by MarcoGorelli** [comment]
Hi @itholic 

Thanks for having reported this - there's now a release candidate for pandas 2.0.0, which you can install with
```
mamba install -c conda-forge/label/pandas_rc pandas==2.0.0rc0
```
or
```
python -m pip install --upgrade --pre pandas==2.0.0rc0
```

If you try it out and report bugs, then we can fix them before the final 2.0.0 release
